packages/browser/src/core/analytics/__tests__/integration.test.ts

```diff
@@ -224,6 +224,19 @@ describe('Analytics', () => {
   })
 
   describe('reset', () => {
+    it('resets anonymousId if run synchronously', async () => {
+      const analytics = new Analytics({ writeKey: '' })
+      const getAnonId = () =>
+        retrieveStoredData({
+          cookieNames: ['ajs_anonymous_id'],
+          localStorageKeys: ['ajs_anonymous_id'],
+        }).ajs_anonymous_id
+      void analytics.track('foo')
+      expect(getAnonId()).toBeTruthy()
+      analytics.reset()
+      expect(getAnonId()).toBeFalsy()
```

Comment:
Since the race condition was that the `track` call would eventually set anonId again, I think checking that it is falsey after the track call completes would cover it (and fail without these changes)

Reply:
I actually moved the test to a different file, tested both failure and success condition!